docs/operate/reference/components/sensor/lorawan.md

```diff
@@ -0,0 +1,519 @@
+  - uses LoRaWAN MAC specification version 1.0.3
+
+Device-specific models for Dragino and Milesight sensors provide default values for `app_key`, `network_s_key`, and `app_s_key`, so you can omit those fields from your configuration.
+
+Configure attributes based on the tables below:
+
+{{% /tab %}}
+{{< /tabs >}}
+
+The following attributes are available for nodes:
+
+<!-- prettier-ignore -->
+| Name | Type | Required? | Description |
+| ---- | ---- | --------- | ----------- |
+| `dev_eui` | string | **Required** | The **device EUI (Extended Unique Identifier)**, a unique 64-bit identifier for the LoRaWAN device in hexadecimal format (16 characters). Found on your device or in device packaging. |
+| `join_type` | string | Optional | The [activation protocol](#activation-protocols) used to secure this network. <br> Default: "OTAA". <br> Options: "OTAA", "ABP" |
+| `app_key` | string | Optional | The 128-bit hexadecimal AES **application key** used for device authentication and session key derivation. Required for OTAA activation protocol. Found in the device datasheet. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `dev_addr` | string | Optional | 32-bit hexadecimal **device address** used to identify this device in uplink messages. Required for ABP activation protocol. Found in the device datasheet or in device packaging. |
+| `app_s_key` | string | Optional | 128-bit hexadecimal **application session key** used to decrypt uplink messages. Required for ABP activation protocol. Found in the device datasheet or in device packaging. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `network_s_key` | string | Optional | 128-bit hexadecimal **network session key** used to decrypt uplink messages. Required for ABP activation protocol. Found in the device datasheet or in device packaging. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `gateways` | string[] | **Required** | Name of the [gateway component](#add-a-gateway) in your Viam configuration. |
+| `uplink_interval_mins` | decimal | **Required** | Interval between uplink messages sent from the node, in minutes. Found in the device datasheet, but can be modified. Configured by downlink after initial connection. <br> On `milesight-ct101`, defaults to `10`. On `milesight-em310-tilt`, defaults to `1080`. |
+| `decoder_path` | string | **Required** | Path to a Javascript **decoder script** used to interpret data transmitted from the node. You can use a local path on your device or an HTTP(S) URL that points to a file on a remote server. If the decoder script provides multiple implementations, uses the Chirpstack version. Not compatible with The Things Network decoders. <br> On supported Milesight and Dragino models, defaults to the latest decoder published by the manufacturer on GitHub. |
```

Comment:
For all the tables, if they do stay in instead of going into readmes, maybe move all the required attributes to the top, and/or make the tables match what's in the readme (if possible?) so that if I have the readme open and this open I don't wonder "hmm are all the same attributes listed in both places or not? I'm cross-checking and kinda overwhelmed!"

Reply:
Trouble is, different modules have different required attributes, and depending on your configuration, certain attributes are required or optional. I've tried to put these in order from most->least likely to be relevant, unfortunately a lot of them are relevant!